src/python/pants/init/plugin_resolver.py

```diff
@@ -107,8 +88,11 @@ def _resolve_exact_plugin_locations(self):
     resolve_hash = hasher.hexdigest()
     resolved_plugins_list = os.path.join(self.plugin_cache_dir, f'plugins-{resolve_hash}.txt')
 
+    if self._plugins_force_resolve:
```

Comment:
Rather than a flag, could we add a lockfile? That would have both reproducibility and speed benefits.

Reply:
That lockfile already exists / is used here. The flag nukes it.

Reply:
Ok. I was thinking of something inside the repository... I think that this is outside, in a cache directory?

Reply:
Yes, in ~/.cache/pants/plugins.  It's keyed off the set of plugins.

Reply:
What would be the advantage of having this in the repo?

Reply:
The same as for #8727: improved reproducibility of builds, due to avoiding any resolution at all happening on a user's machine: just fetching.

The convention for "lock files" in all languages/resolvers that I have encountered is to commit them.